src/datasets/data_files.py

```diff
@@ -235,20 +240,20 @@ def _get_data_files_patterns(pattern_resolver: Callable[[str], List[PurePath]])
     for patterns_dict in ALL_DEFAULT_PATTERNS:
         non_empty_splits = []
         for split, patterns in patterns_dict.items():
-            try:
-                for pattern in patterns:
+            for pattern in patterns:
+                try:
                     data_files = pattern_resolver(pattern)
-                    if len(data_files) > 0:
-                        non_empty_splits.append(split)
-                        break
-            except FileNotFoundError:
-                pass
+                except FileNotFoundError:
+                    continue
+                if len(data_files) > 0:
+                    non_empty_splits.append(split)
+                    break
```

Comment:
this is needed because now `resolve_pattern` returns FileNotFoundError if it can't resolve at least one file